From f7430355445b4852fa43a201239ae5fc463f387d Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Tue, 4 Jun 2019 14:45:22 +0200
Subject: [PATCH] PONRTSYS-4221: Reset the XGMAC for the PON IP

When the system is initialized, reset the XGMAC connected to the PON IP.
Without this reset the downstream from the PON IP to the GSWIP will not
work. This is executed when the PON Ethernet driver is loaded.
---
 drivers/net/datapath/dpm/gswip31/datapath_misc.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/datapath/dpm/gswip31/datapath_misc.c b/drivers/net/datapath/dpm/gswip31/datapath_misc.c
index 2bc6480ada4a..98d66705a70c 100644
--- a/drivers/net/datapath/dpm/gswip31/datapath_misc.c
+++ b/drivers/net/datapath/dpm/gswip31/datapath_misc.c
@@ -1352,6 +1352,11 @@ static int pon_config(int inst, int ep, struct dp_port_data *data, u32 flags)
 	 */
 	mac_ops->mac_op_cfg(mac_ops, RX_TIME_NO_INSERT);
 
+	/* Reset the MAC, without this reset the downstream from the PON IP
+	 * will not work when the MAC is not reset in U-Boot before.
+	 */
+	mac_ops->soft_restart(mac_ops);
+
 	return 0;
 }
 
